projects/ngx-datatable/src/lib/components/columns/column.directive.ts

```diff
@@ -4,19 +4,18 @@ import {
   Input,
   OnChanges,
   PipeTransform,
-  SimpleChanges,
   TemplateRef
 } from '@angular/core';
 import { DataTableColumnHeaderDirective } from './column-header.directive';
 import { DataTableColumnCellDirective } from './column-cell.directive';
 import { DataTableColumnCellTreeToggle } from './tree.directive';
 import { ColumnChangesService } from '../../services/column-changes.service';
-import { TableColumnProp } from '../../types/table-column.type';
+import { TableColumn, TableColumnProp } from '../../types/table-column.type';
 import { DataTableColumnGhostCellDirective } from './column-ghost-cell.directive';
 import { CellContext, HeaderCellContext } from '../../types/public.types';
 
 @Directive({ selector: 'ngx-datatable-column' })
-export class DataTableColumnDirective<TRow> implements OnChanges {
+export class DataTableColumnDirective<TRow> implements TableColumn, OnChanges {
```

Comment:
I do not understand why this directive needs to implement the TableColumn. I removed it and it works as is.

Reply:
`DataTableColumnDirective` is converted to a `TableColumn` buy just copying the properties. See https://github.com/siemens/ngx-datatable/blob/master/projects/ngx-datatable/src/lib/components/datatable.component.ts#L823 and https://github.com/siemens/ngx-datatable/blob/master/projects/ngx-datatable/src/lib/utils/column-helper.ts#L87 

If we wan to prevent future mistakes here, we should make sure, that `DataTableColumnDirective` is a `TableColumn`. Otherwise one may add new properties, that conflict the `TableColumn`. It will always work at compile time when removing this, but you may get runtime errors.